### PR TITLE
bump Babylon contracts

### PR DESCRIPTION
**Pull request by SebastianElvis**
TODO before merging

- [x] wait until https://github.com/babylonchain/babylon-contract/pull/163 and https://github.com/babylonchain/babylon-contract/pull/162 are merged

### CODE REVIEW

File: x/babylon/keeper/wasm.go [hunk 1]
```diff
@@ -23,10 +23,7 @@ func (k Keeper) SendBeginBlockMsg(ctx sdk.Context) error {
 	headerInfo := ctx.HeaderInfo()
 	msg := contract.SudoMsg{
 		BeginBlockMsg: &contract.BeginBlock{
```
Comment (maurolacy) [inline x/babylon/keeper/wasm.go:25]: There's no `BeginBlockMsg` anymore. Values are just flattened inside the `SudoMsg::BeginBlock`.

The same with`SudoMsg::EndBlock`.

**Review by maurolacy — COMMENTED** [review]
I think this still requires adjustment, as one level of nesting was removed from the begin / end block sudo defs.

File: x/babylon/keeper/wasm.go (hunk 1 above)
Comment (SebastianElvis) [reply x/babylon/keeper/wasm.go:25]: Tried, and looks like the marshaler already did flattening for us. Now I changed the sudo msg def to 

```go
// SudoMsg is a message sent from the Babylon module to a smart contract
type SudoMsg struct {
	BeginBlockMsg struct {
		HashHex    string `json:"hash_hex"`     // HashHex is the hash of the block in hex
		AppHashHex string `json:"app_hash_hex"` // AppHashHex is the app hash of the block in hex
	} `json:"begin_block,omitempty"`
	EndBlockMsg struct {
		HashHex    string `json:"hash_hex"`     // HashHex is the hash of the block in hex
		AppHashHex string `json:"app_hash_hex"` // AppHashHex is the app hash of the block in hex
	} `json:"end_block,omitempty"`
}
```

and construct the sudo msg accordingly. The e2e test gives the following 

```
--- FAIL: TestBabylonSDKTestSuite (1.75s)
    --- FAIL: TestBabylonSDKTestSuite/Test3BeginBlock (0.00s)
        main_test.go:134: 
                Error Trace:    /home/runchao/Projects/Babylon/babylon-sdk/tests/e2e/main_test.go:134
                Error:          Received unexpected error:
                                Error parsing into type babylon_apis::btc_staking_api::SudoMsg: Expected this character to start a JSON value.: execute wasm contract failed
                Test:           TestBabylonSDKTestSuite/Test3BeginBlock
    --- FAIL: TestBabylonSDKTestSuite/Test4EndBlock (0.00s)
        main_test.go:140: 
                Error Trace:    /home/runchao/Projects/Babylon/babylon-sdk/tests/e2e/main_test.go:140
                Error:          Received unexpected error:
                                Error parsing into type babylon_apis::btc_staking_api::SudoMsg: Expected this character to start a JSON value.: execute wasm contract failed
                Test:           TestBabylonSDKTestSuite/Test4EndBlock
```

See https://github.com/babylonchain/babylon-sdk/tree/bump-contract-test-flattening for the detail

So I suggest we keep the struct as is.

Comment (maurolacy) [reply x/babylon/keeper/wasm.go:25]: I see, thanks for testing it.

File: tests/testdata/btc_staking.wasm
Comment (maurolacy) [inline tests/testdata/btc_staking.wasm:1]: You can now update these blobs to the just released v0.7.0 tag. https://github.com/babylonchain/babylon-contract/releases/tag/v0.7.0-rc.0

**Review by maurolacy — APPROVED** [review]

File: tests/testdata/version.txt [hunk 2]
```diff
@@ -1 +1 @@
-938124458cdfdcd1b7b6bf14762ef606447c3c89
+ed2bb75d109c579ad9b1e8ad1efe089421326028
```
Comment (maurolacy) [inline tests/testdata/version.txt:1]: Nit: You can put the version tag here. The `download_releases.sh` script does that for you. Will check that it works.

**Comment by maurolacy** [comment]
CI is not passing, btw.

File: tests/testdata/version.txt (hunk 2 above)
Comment (maurolacy) [reply tests/testdata/version.txt:1]: The `download_releases.sh` script works fine. I run it like this:

`GITHUB_API_TOKEN="$BBN_GITHUB_API_TOKEN" ./download_releases.sh v0.7.0-rc.0`

Where `BBN_GITHUB_API_TOKEN` contains a token with access to the `babylon-contract` repo.

**Comment by SebastianElvis** [comment]
> CI is not passing, btw.

CI for this repo is broken as github action could not fetch private repositories. Asked infra team for help and they will do sth depending on the priority

**Pull request merged**